### PR TITLE
SHO-36: Perk Info

### PR DESCRIPTION
**Pull request by kevinochoag**
## Describe your changes

This PR adds a new component: Section Info.

## Task URL (Asana, Jira, etc...)

[SHO-36](https://cultoperrocafe.nifty.pm/l/7nZ8WlIDh3BHT)

## What was done in this pull request

- [x] Added new component: `<SectionInfo />`.
- [x] Added Storybook variants.

## Demo / Screenshot / Video

sho-36

### CODE REVIEW

File: app/components/molecules/section-info/section-info.component.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import ingredients from './img/ingredients.svg';
+import internet from './img/internet.svg';
+import {LeadingIcon, Props, StyledSectionInfo} from './section-info.styles';
+import {Hr} from '../../atoms/hr';
+import {Paragraph} from '../../atoms/paragraph';
+
+export const SectionInfo: FC<Props> = ({children, variant = 'none'}) => {
+  let leadingIcon;
+
+  switch (variant) {
```
Comment (jmsalcido) [inline app/components/molecules/section-info/section-info.component.tsx]: why is this switch needed?

wouldnt this work?

```typescript
<LeadingIcon src={`./img/{variant}.svg`} />
```

File: app/components/molecules/section-info/section-info.component.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      leadingIcon = events;
+      break;
+    default:
+      break;
+  }
+
+  return (
+    <StyledSectionInfo>
+      {leadingIcon && <LeadingIcon src={leadingIcon} />}
+      <Hr></Hr>
```
Comment (jmsalcido) [inline app/components/molecules/section-info/section-info.component.tsx]: `<Hr/>`

File: app/components/molecules/section-info/section-info.component.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import events from './img/events.svg';
+import games from './img/games.svg';
+import ingredients from './img/ingredients.svg';
+import internet from './img/internet.svg';
+import {LeadingIcon, Props, StyledSectionInfo} from './section-info.styles';
+import {Hr} from '../../atoms/hr';
+import {Paragraph} from '../../atoms/paragraph';
+
+export const SectionInfo: FC<Props> = ({children, variant = 'none'}) => {
```
Comment (jmsalcido) [inline app/components/molecules/section-info/section-info.component.tsx]: why make the default variant `'none'` if there is no `none` variant?, are we reusing this component in another part?

AFAIK those are the variants:
image

Just accept the value as it is, none is undefined

File: app/components/atoms/paragraph/paragraph.component.tsx [hunk 4]
```diff
@@ -9,8 +9,9 @@ interface Props {
 export const Paragraph = styled.p<Props>`
   color: ${({theme}) => theme.colors.black};
   font-family: ${({theme}) => theme.fonts.body};
-
   ${({bold}) => bold && 'font-weight: bold'};
   ${({italic}) => italic && 'font-style: italic'};
   ${({lineHeight}) => lineHeight && `line-height: ${lineHeight}`};
+  margin-block-start: 0;
```
Comment (jmsalcido) [inline app/components/atoms/paragraph/paragraph.component.tsx:15]: why is this needed?

File: app/components/molecules/section-info/section-info.styles.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import styled from 'styled-components';
+
+export interface Props {
+  variant: 'none' | 'internet' | 'events' | 'ingredients' | 'games';
+  children: string;
```
Comment (jmsalcido) [inline app/components/molecules/section-info/section-info.styles.tsx]: isnt `children` a prop already defined by default? - or are we just explicitly doing a definition to be string here? that's ok.

File: app/components/molecules/section-info/section-info.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import {ComponentMeta, ComponentStory} from '@storybook/react';
+
+import {SectionInfo} from './section-info.component';
+
+export default {
+  title: 'Molecules/SectionInfo',
+  component: SectionInfo,
+} as ComponentMeta<typeof SectionInfo>;
+
+const Template: ComponentStory<typeof SectionInfo> = (args) => (
```
Comment (jmsalcido) [inline app/components/molecules/section-info/section-info.stories.tsx]: Made a fast change:

```typescript
import {ComponentMeta, ComponentStory} from '@storybook/react';

import {SectionInfo} from './section-info.component';

export default {
  title: 'Molecules/SectionInfo',
  component: SectionInfo,
} as ComponentMeta<typeof SectionInfo>;

const Template: ComponentStory<typeof SectionInfo> = (props) => (
  <SectionInfo variant={props.variant}>
    {props.children}
  </SectionInfo>
);

export const Internet = Template.bind({});
Internet.args = {
  variant: 'internet',
  children: 'Internet rápido',
};

export const Ingredients = Template.bind({});
Ingredients.args = {
  variant: 'ingredients',
  children: 'Ingredientes 100% naturales',
};

export const Games = Template.bind({});
Games.args = {
  variant: 'games',
  children: 'Juegos de mesa',
};

export const Events = Template.bind({});
Events.args = {
  variant: 'events',
  children: 'Hosteo de eventos',
};

```


Look at the usage:

```typescript
<SectionInfo variant='ingredients'>
    Ingredientes 100% naturales
  </SectionInfo>
```

so if... I did this:

```typescript
<SectionInfo variant='games'>
    Ingredientes 100% naturales
  </SectionInfo>
```

It would look like this:

image

So why not the components are actually `<GameInfo>Information</GameInfo>`, `InternetInfo`, `EventsInfo` instead of `<SectionInfo variant="foo">Information</SectionInfo>`

File: app/components/atoms/paragraph/paragraph.component.tsx (hunk 4 above)
Comment (kevinochoag) [reply app/components/atoms/paragraph/paragraph.component.tsx:15]: image
Yes!. For no reason, the margin-block-start and margin-block-end has a 1em value as default value.

File: app/components/molecules/section-info/section-info.styles.tsx (hunk 5 above)
Comment (kevinochoag) [reply app/components/molecules/section-info/section-info.styles.tsx]: Yes! that's right

Comment (jmsalcido) [reply app/components/molecules/section-info/section-info.styles.tsx]: nah https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example make it ReactNode please

File: app/components/atoms/paragraph/paragraph.component.tsx (hunk 4 above)
Comment (jmsalcido) [reply app/components/atoms/paragraph/paragraph.component.tsx:15]: this would affect all `<Paragraph>` components FYI.

Comment (kevinochoag) [reply app/components/atoms/paragraph/paragraph.component.tsx:15]: Yes I know, I told Saúl and he tells me that he hasn't used the Paragraph yet. So it will not affect the created components

File: app/components/molecules/section-info/section-info.stories.tsx (hunk 6 above)
Comment (kevinochoag) [reply app/components/molecules/section-info/section-info.stories.tsx]: image
It works, but I don't how to use multiple components in one storybook file, I need to create another .stories for each component?

File: app/components/molecules/section-info/section-info.component.tsx (hunk 1 above)
Comment (kevinochoag) [reply app/components/molecules/section-info/section-info.component.tsx]: image
It works! I I ¿already deleted the use of the switch

File: app/components/molecules/section-info/section-info.component.tsx (hunk 2 above)
Comment (kevinochoag) [reply app/components/molecules/section-info/section-info.component.tsx]: image
Done

**Review by tonymtz — CHANGES_REQUESTED** [review]
Images are not working :(

File: app/components/molecules/section-info/section-info.component.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import events from './img/events.svg';
+import games from './img/games.svg';
+import ingredients from './img/ingredients.svg';
+import internet from './img/internet.svg';
+import {LeadingIcon, Props, StyledSectionInfo} from './section-info.styles';
+import {Paragraph} from '../../atoms/paragraph';
+
+export const InternetInfo = () => {
+  return (
+    <>
```
Comment (tonymtz) [inline app/components/molecules/section-info/section-info.component.tsx:12]: No wrapper, interesting. Bookmark this comment for later :)

**Review by tonymtz — DISMISSED** [review]

**Review by jmsalcido — CHANGES_REQUESTED** [review]
**Deployment failed:**
https://b0agvtnsv-3a3ad3a7297dea854149.myshopify.dev/

Something is wrong in this branch

**Comment by jmsalcido** [comment]
any update on this?

**Comment by kevinochoag** [comment]
I don't have an idea. I'm fixing this in another branch [SHO-36-FIX](https://github.com/tepachelabs/perro-cafe-storefront/pull/39)
